src/core/io/pla_parser.cpp

```diff
@@ -17,10 +17,16 @@ namespace syrec {
 
             if (line.rfind(".i", 0) == 0) {
                 nInputs = std::stoi(line.substr(3));
+                // reset and resize the tt constants.
+                tt.resetConstants();
+                tt.resizeConstants(nInputs);
             }
 
             else if (line.rfind(".o", 0) == 0) {
                 nOutputs = std::stoi(line.substr(3));
+                // reset and resize the tt garbage.
+                tt.resetGarbage();
```

Comment:
the `parsePLA` does not reset the underlying TT as of now. With these changes, it does, however, reset the constants and garbage values.
I'd expect this function to be only called on a "fresh" TT, so for that use case, no reset is needed at all.
If you really want to be on the safe side, then provide a `reset` method in the TT class that resets all internal datastructures and put `tt.reset()` at the top of this function.